crates/nu-protocol/src/parse_warning.rs

```diff
@@ -3,6 +3,7 @@ use miette::Diagnostic;
 use serde::{Deserialize, Serialize};
 use thiserror::Error;
 
+#[allow(dead_code)]
```

Comment:
it's a `pub enum`, not sure why we need this :thinking:

Reply:
Oh, I thought clippy will warn about this because we don't use ParseWarning inside nushell.

But after I remove it and run clippy, it pass :)
Good catch!